mkosi/__init__.py

```diff
@@ -4948,6 +4952,22 @@ def run_build(
         )
 
 
+def ensure_tools_tree_has_etc_resolv_conf(config: Config) -> None:
+    if not config.tools_tree:
+        return
+
+    # We can't bind mount in the hosts's /etc/resolv.conf if this file doesn't exist without making the
+    # entirety of /etc writable or messing around with overlayfs, so let's just ensure it exists.
+    path = config.tools_tree / "etc/resolv.conf"
+
+    if not path.is_symlink() and not path.exists():
```

Comment:
This bends my brain a bit. Logically this is equivalent to `not (path.is_symlink() or path.exists())`. `Path.exists()` by default follows symlinks, so this covers the case when `path` is a dangling symlink, since `Path.is_symlink` is also true for them, right? I think in that case this could just be
```suggestion
    if not path.exists(follow_symlinks=False):
```
to ensure that there's something to mount over.

Reply:
So as mentioned in chat, follow_symlinks unfortunately doesn't exist for exists() until 3.12